dandi/metadata.py

```diff
@@ -90,9 +90,86 @@ def get_metadata(path):
     return meta
 
 
+def _parse_iso8601(age):
+    """checking if age is proper iso8601, additional formatting"""
+    # allowing for comma instead of ., e.g. P1,5D
+    age = age.replace(",", ".")
+    pattern = (
+        "^P(?!$)(\\d+(?:\\.\\d+)?Y)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?W)?(\\d+(?:\\.\\d+)?D)?"
+        "(T(?=\\d)(\\d+(?:\\.\\d+)?H)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?S)?)?$"
+    )
+
+    matchstr = re.match(pattern, age, flags=re.I)
+    if matchstr:
+        age_rem = age.replace(matchstr.group(0), "")
+        if age_rem:
+            raise ValueError(
+                f"ISO 8601 expected, but {age} was received,"
+                f"no rules to parse {age_rem}"
+            )
```

Comment:
if you could add a test to cover this exception raise so whenever we change code and rename some variables exception raising doesn't raise some formatting exception?

Reply:
at the end I'm not sure what case I was trying to cover here... I guess all wrong values go to the next exception (that is tested)